### PR TITLE
Force hipify to use copied version from rocm-6.3.0-14776 build

### PR DESCRIPTION
**Pull request by TedThemistokleous**
### Description

Copied out hipifiy-perl from working docker container as mentioned by @xinyazhang 
Retargeted amd_hipify to use the copied item found in onnxruntime root

### Motivation and Context

Changed as an upstream change to the rocm stack requires a bunch of work required to get a working rebuild of ROCm EP.
Currently this inhibits development and blocks 6.3 wheel builds

### CODE REVIEW

**Review by causten — COMMENTED** [review]
Does this need to be merged upstream?  Would the path for hipify-perl be appropriate? Seems like you are suggesting it belongs in the root.

**Review by xinyazhang — CHANGES_REQUESTED** [review]
`hipify-perl` should be placed under the same directory of `amd_hipify.py` rather than root of the source tree, unless there is really good reason.

**Comment by xinyazhang** [comment]
> Does this need to be merged upstream? Would the path for hipify-perl be appropriate? Seems like you are suggesting it belongs in the root.

This is a temporary solution and we are working on another PR that eliminates `-roc` argument from https://github.com/ROCm/onnxruntime/blob/2580d935cbecd756cef435fb173a2f10237e9d2a/tools/ci_build/amd_hipify.py#L14
, which should work for newer hipify-perl

**Comment by TedThemistokleous** [comment]
> `hipify-perl` should be placed under the same directory of `amd_hipify.py` rather than root of the source tree, unless there is really good reason.

Done. Just retesting this. and will update once done.

**Comment by TedThemistokleous** [comment]
Moved it and had to update permissions. Looks like its built and working minus some other test thats failed

image

File: tools/ci_build/amd_hipify.py [hunk 1]
```diff
@@ -205,4 +205,4 @@ def hipify(hipify_perl_path, src_file_path, dst_file_path):
     parser.add_argument("src", help="src")
     args = parser.parse_args()
 
-    hipify(args.hipify_perl, args.src, args.output)
+    hipify("/onnxruntime/tools/ci_build/hipify-perl", args.src, args.output)
```
Comment (xinyazhang) [inline tools/ci_build/amd_hipify.py:208]: Use
``` python
hipify(os.path.join(os.path.dirname(__file__), "hipify-perl"), args.src, args.output)
```
Instead

**Review by xinyazhang — CHANGES_REQUESTED** [review]
Do not use absolute path. Compute it in your `amd_hipify.py`

File: tools/ci_build/amd_hipify.py (hunk 1 above)
Comment (TedThemistokleous) [reply tools/ci_build/amd_hipify.py]: Done

**Review by xinyazhang — APPROVED** [review]

**Pull request merged**